src/main/scripts/src/app/core/rest/collection.service.ts

```diff
@@ -62,4 +70,15 @@ export class CollectionService {
     return `/lumeer-engine/rest/organizations/${organization}/projects/${project}/collections/`;
   }
 
+  private static handleError(error: HttpErrorResponse) {
+    if (error.status === 400) {
+      return Observable.throw(new BadInputError('Name already exists'));
```

Comment:
Do we really want to show this error message in all method calls of this service? It does not say much since name does not have to be unique. If we want to handle such errors, we should probably also implement async form validation calling remote service.